.github/workflows/release.yml

```diff
@@ -83,40 +83,12 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: "${{ env.PYTHON_MAX_VERSION }}"
-      - name: Look up the related GitHub PR branch name
-        # Need GH Token to use gh api
-        env:
-          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          pip install requests
-          echo "GH_PR_BRANCH=$(python scripts/get_release_branch.py)" >> $GITHUB_ENV
-      - name: Ensure that version tag matches branch version
-        env:
-          GH_PR_BRANCH: ${{ env.GH_PR_BRANCH }}
-        run: |
-          BRANCH_VERSION=$(echo "$GH_PR_BRANCH" | sed 's/release\///' )
-
-          if [ "$BRANCH_VERSION" != "$GIT_TAG" ]
-          then
-            echo "ERROR: Version number from tag does not match the version number from the branch name."
-            exit 1
-          fi
```

Comment:
There's a part of me that thinks. Oh good the PR check won't block release, but I want to just make sure we still create the PR to get it merged. Thoughts?

Reply:
Ah, good point. I think we'll want this check just to make sure that the PR exists. Added it back with some stage names / error messages tweaked